.github/workflows/npm-publish.yml

```diff
@@ -1,15 +1,18 @@
-on: push
+on:
+  release:
+    types:
+      - published
 
 jobs:
   publish:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1
-      - uses: actions/setup-node@v1
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
         with:
-          node-version: 12
+          node-version: 22.9
```

Comment:
Have you used matrix values before?
You can have the tests run with multiple node versions to make sure it works with the latest and the oldest (i think 18 will be the oldest we support here)

Reply:
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow

Reply:
Oh, this is the `publish` step... this should only be 1 node version sorry.


We do also want a test step that runs tests - that should use a version matrix